Index: linux-6.12.50/arch/arm64/boot/dts/qcom/msm8916-generic-sp9070.dts
===================================================================
--- /dev/null
+++ linux-6.12.50/arch/arm64/boot/dts/qcom/msm8916-generic-sp9070.dts
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include "msm8916-ufi.dtsi"
+
+/ {
+	model = "Generic SP9070 4G LTE WiFi Modem";
+	compatible = "generic,sp9070", "qcom,msm8916";
+};
+
+&button_restart {
+	gpios = <&tlmm 107 GPIO_ACTIVE_LOW>;
+};
+
+&led_b {
+	gpios = <&tlmm 28 GPIO_ACTIVE_HIGH>;
+};
+
+&led_g {
+	gpios = <&tlmm 10 GPIO_ACTIVE_HIGH>;
+};
+
+&led_r {
+	gpios = <&tlmm 9 GPIO_ACTIVE_HIGH>;
+};
+
+&button_default {
+	pins = "gpio107";
+	bias-pull-up;
+};
+
+&gpio_leds_default {
+	pins = "gpio28", "gpio10", "gpio9";
+};
